packages/terra-clinical-onset-picker/src/OnsetPicker.jsx

```diff
@@ -0,0 +1,362 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import 'terra-base/lib/baseStyles';
+import Field from 'terra-form/lib/Field';
+import Fieldset from 'terra-form/lib/Fieldset';
+import NumberField from 'terra-form/lib/NumberField';
+import DatePicker from 'terra-date-picker';
+import SelectField from 'terra-form/lib/SelectField';
+import OnsetUtil from './OnsetUtil';
+
+const propTypes = {
+  /**
+   * The ISO 8601 string representation of the birth date to calculate an onset date for the AGE precision.
+   * Also limits the earliest possible date that can be selected for an onset date for YEAR, MONTH, and DATE precision.
+   */
+  birthdate: PropTypes.string.isRequired,
+
+  /**
+   * The granularity of the onset date. AGE, YEAR, MONTH, and DATE are accepted.
+   */
+  granularity: PropTypes.oneOf(['AGE', 'YEAR', 'MONTH', 'DATE']),
+
+  /**
+   * Name of the granularity select. The name should be unique.
+   */
+  granularitySelectName: PropTypes.string.isRequired,
+
+  /**
+   * A callback function to execute when a granularity is selected.
+   * The first parameter is the changed granularity value.
+   */
+  granularitySelectOnChange: PropTypes.func,
+
+  /**
+   * The precision of the onset date. This value should be in the set of precisions passed to the precisionSet prop.
+   */
+  precision: PropTypes.oneOf(['ON/AT', 'ABOUT', 'BEFORE', 'AFTER', 'UNKNOWN']),
```

Comment:
It might be valuable to create a const array of these precision options that we can pass here, into precisionSet and to set as the precisionSet's default value. I'd then expect us to export this array of options as `OnsetPicker.PRECISION_OPTIONS` or a similar name.

Reply:
OnsetPicker.PrecisionOptions enums created.